### ISSUE TITLE
quickfix list / location list improvements

### ISSUE THREAD

**Issue by dlants** [neovim/neovim#9188]
Plugins that provide code intelligence for various languages often use the quickfix list for global project errors, and the location list for live-editing errors.

Right now, vim's lists are limited in two ways:

1. There's a maximum length (I believe 4096 bytes), after which characters are dropped.
2. Each item in the list takes up a single line.

This is not ideal for displaying error messages, as they can be quite long (when providing context, like type information), and formatted.

Some language plugins work around this issue by parsing multi-line error messages into multiple list entries, but then you create another issue where some lines in the list do not correspond to error locations, and moving through the list becomes awkward.

It would be really nice to remove the line length constraint on list items, and to allow them to provide formatting (mostly, line breaks).

**Comment by justinmk** [comment]
>  would be really nice to remove the line length constraint on list items, and to allow them to provide formatting

Who's going to provide that formatting, which the very tool that produced the output couldn't bother to do?

This is a "make things better" request without a clear actionable resolution. I don't see how megabyte-sized unbroken lines is going to help anything.

**Comment by dlants** [comment]
> This is a "make things better" request without a clear actionable resolution.
The resolution would be to remove the line length limit, and to show newlines within quickfix/location list items.

> Who's going to provide that formatting, which the very tool that produced the output couldn't bother to do?
Plugin authors currently remove the formatting provided by compilers in order to make them fit into the location list format, as well as trim them to length.

> In any case if this can be improved it's something that Vim will spend time on, not something Nvim has any reason to diverge on.
I think this would be a very nice quality-of-life improvement for plugin authors -- right now providing context about errors in vim is hard, and is something that pretty much no plugin does well. Seems to jive with the "unblock third parties", "prefer usability" goals of the project.

Apologies if the text in the initial issue was not clear!

**Comment by justinmk** [comment]
> Seems to jive with the "unblock third parties", "prefer usability" goals of the project.

quickfix enhancements are not in any danger of slowing down, @yegappan has been doing a lot of work there. Nvim merges all of those, and there's no reason for Nvim to spend time on it.
